modules/core/exported/client.go

```diff
@@ -40,17 +40,56 @@ type ClientState interface {
 	GetLatestHeight() Height
 	Validate() error
 
+	// Status must return the status of the client. Only Active clients are allowed to process packets.
+	Status(ctx sdk.Context, clientStore sdk.KVStore, cdc codec.BinaryCodec) Status
```

Comment:
reorganized the ordering:
- client utiltity funcs
- initialize
- proof verification
- update functions
- proposal
- upgrade